apps/editing-toolkit/editing-toolkit-plugin/coming-soon/test/class-coming-soon-test.php

```diff
@@ -154,4 +188,73 @@ public function test_add_option_to_new_site_without_coming_soon_meta() {
 		do_action( 'wpmu_new_blog', get_current_blog_id(), null, null, null, null, $meta );
 		$this->assertFalse( get_option( 'wpcom_public_coming_soon' ) );
 	}
+
+	/**
+	 * Tests that is_accessed_by_valid_share_link() returns true for requests
+	 * that include valid preview link.
+	 *
+	 * @return void
+	 */
+	public function test_is_accessed_by_valid_share_link_success() {
+		update_option( 'wpcom_public_preview_links', self::$preview_links );
+
+		$result = is_accessed_by_valid_share_link( self::$preview_links[0]['code'] );
+
+		$this->assertTrue( $result );
+	}
+
+	/**
+	 * Tests that is_accessed_by_valid_share_link() returns false for requests
+	 * that include invalid preview link.
+	 *
+	 * @return void
+	 */
+	public function test_is_accessed_by_valid_share_link_failure() {
+		update_option( 'wpcom_public_preview_links', self::$preview_links );
+
+		$result = is_accessed_by_valid_share_link( 'foo-bar' );
+
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * Tests if get_share_code() return share code from GET param.
+	 *
+	 * @return void
+	 */
+	public function test_get_share_code_gets_code_from_get() {
+		self::set_valid_share_code_get_parameter();
+
+		$share_code = get_share_code();
+
+		$this->assertEquals( 'sharing-code', $share_code );
+
+		self::delete_preview_links_parameters();
```

Comment:
Should this happen on `::tearDown()` so it gets called even when the test fails?

Reply:
Makes sense, I've changed it.